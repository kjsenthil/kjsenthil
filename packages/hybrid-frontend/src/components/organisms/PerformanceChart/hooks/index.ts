/* eslint-disable import/prefer-default-export */
export { usePerformanceChartDimension } from './usePerformanceChartDimension';
