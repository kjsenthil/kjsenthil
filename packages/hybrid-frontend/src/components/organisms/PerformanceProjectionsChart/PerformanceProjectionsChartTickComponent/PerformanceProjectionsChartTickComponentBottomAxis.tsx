import * as React from 'react';
import { TickRendererProps } from '@visx/axis';
import { Text } from '@visx/text';
import { usePerformanceProjectionsChartStyles } from '../performanceProjectionsChartStyles/performanceProjectionsChartStyles';

export interface PerformanceProjectionsChartTickComponentBottomAxisProps extends TickRendererProps {
  // Note: we have to pass chartStyles as a prop rather than having
  // PerformanceChartTickComponent uses useChartStyles() directly. Otherwise we
  // will run into the mismatched hook call order error when the chart period
  // changes.
  // The cause for this is likely due to how d3 is handling the showing / hiding
  // of chart ticks. Unknown for sure at this point but passing the chartStyles
  // down as a prop works well enough.
  chartStyles: ReturnType<typeof usePerformanceProjectionsChartStyles>;

  todayAge: number;
}

/**
 * The formatted value passed to our tick component will be a full year string.
 * This function adds an age text to the formatted value.
 */
export function getAgeText(formattedValue: string | undefined, todayAge: number): string {
  const tickYear = Number(formattedValue);

  if (Number.isNaN(tickYear)) return `Age unknown`;

  const todayYear = new Date().getFullYear();

  const age = todayAge + (tickYear - todayYear);

  return `Age ${age}`;
}

export default function PerformanceProjectionsChartTickComponentBottomAxis({
  chartStyles,
  todayAge,
  x,
  y,
  formattedValue,
  ...tickLabelProps
}: PerformanceProjectionsChartTickComponentBottomAxisProps) {
  return (
    <>
      <Text
        x={x}
        y={y}
        {...tickLabelProps}
        fill={chartStyles.TEXT_COLOR.AXES}
        fontSize={chartStyles.TEXT_SIZE.AXES}
        fontFamily={chartStyles.TEXT_FONT.COMMON}
        width={56}
        lineHeight={15}
        style={{ fontWeight: 'bold' }}
      >
        {formattedValue}
      </Text>
      <Text
        x={x}
        y={y + 15}
        {...tickLabelProps}
        fill={chartStyles.TEXT_COLOR.AXES}
        fontSize={chartStyles.TEXT_SIZE.AXES}
        fontFamily={chartStyles.TEXT_FONT.COMMON}
        width={56}
        lineHeight={15}
        style={{ fontWeight: 'bold' }}
      >
        {getAgeText(formattedValue, todayAge)}
      </Text>
    </>
  );
}
