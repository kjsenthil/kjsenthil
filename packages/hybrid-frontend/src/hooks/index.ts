import { BasicInfo } from './useBasicInfo';

export { default as useGoals } from './useGoals';
export { default as useBasicInfo } from './useBasicInfo';
export { default as useGoalImages } from './useGoalImages';
export { default as useScreenSize } from './useScreenSize';
export { default as useChartStyles } from './useChartStyles';
export { default as useFeatureFlag } from './useFeatureFlag';
export { default as usePerformanceData } from './usePerformanceData';
export { default as useTimeValueScales } from './useTimeValueScales';
export { default as useStateIsAvailable } from './useStateIsAvailable';
export { default as useContributionsData } from './useContributionsData';
export { default as useGoalsDataForChart } from './useGoalsDataForChart';
export { default as useProjectionsChartData } from './useProjectionsChartData';
export { default as useDispatchThunkOnRender } from './useDispatchThunkOnRender';
export { default as usePerformanceDataPeriod } from './usePerformanceDataPeriod';
export { default as useProjectionsMetadataForProjectionsChart } from './useProjectionsMetadataForProjectionsChart';
export { default as useProjectionsTargetDataForProjectionsChart } from './useProjectionsTargetDataForProjectionsChart';
export { default as useCurrentProjectionsDataForProjectionsChart } from './useCurrentProjectionsDataForProjectionsChart';

export type { BasicInfo };
