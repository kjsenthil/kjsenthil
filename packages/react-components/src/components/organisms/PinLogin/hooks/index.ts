export { default as usePinInputs } from './usePinInputs';

export type { PinInput, PinInputs } from './usePinInputs';
