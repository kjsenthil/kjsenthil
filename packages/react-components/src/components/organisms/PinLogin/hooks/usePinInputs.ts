import React, { useEffect, useRef, useState } from 'react';
import { PinLoginItem } from '../../../../services';
import randomPinIndices from '../../../../services/auth/utils/randomPinIndices';

export interface PinInput extends PinLoginItem {
  ref: React.RefObject<HTMLInputElement | undefined>;
}

export interface PinInputs {
  inputs: PinInput[];
  setInputValue: (index: number, value: number) => void;
  focusInput: (index: number) => void;
}

const usePinInputs = (): PinInputs => {
  const [inputs, setInputs] = useState<PinLoginItem[]>(randomPinIndices());
  const [shouldFocus, setShouldFocus] = useState<boolean>(true);

  const inputRefs = [
    useRef<HTMLInputElement>(),
    useRef<HTMLInputElement>(),
    useRef<HTMLInputElement>(),
  ];

  const setInputValue = (index: number, value: number) => {
    setInputs((currInputs) => {
      const newCurrInputs = currInputs.map((input) => ({ ...input }));
      newCurrInputs[index].value = value;
      return newCurrInputs;
    });
  };

  const focusInput = (index: number) => {
    inputRefs[index].current?.focus();
  };

  useEffect(() => {
    if (shouldFocus && inputRefs[0].current) {
      inputRefs[0].current.focus();
      setShouldFocus(false);
    }
  }, [inputRefs, shouldFocus]);

  return {
    inputs: inputs.map((loginItem, index) => ({ ...loginItem, ref: inputRefs[index] })),
    setInputValue,
    focusInput,
  };
};

export default usePinInputs;
