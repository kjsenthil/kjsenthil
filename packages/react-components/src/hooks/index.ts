export { default as useBreakpoint } from './useBreakpoint';
export { default as useChartStyles } from './useChartStyles';
export { default as useScreenSize } from './useScreenSize';
export { default as useTimeValueScales } from './useTimeValueScales';
