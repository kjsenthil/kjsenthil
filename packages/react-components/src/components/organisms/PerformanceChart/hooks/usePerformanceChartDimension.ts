import * as React from 'react';
import { ChartDimension, ChartDimensionWithExtras, ScreenSize } from '../../../../config/chart';
import { getChartDimensionWithExtras } from '../../../../utils/chart';
import { useScreenSize } from '../../../../hooks';

const basePerformanceChartDimension: ChartDimension = {
  width: 0,
  height: 240,
  margin: {
    top: 0,
    right: 0,
    bottom: 30,
    left: 60,
  },
};

const performanceChartDimensionByScreenSize: Record<ScreenSize, ChartDimension> = {
  [ScreenSize.MOBILE]: basePerformanceChartDimension,
  [ScreenSize.TABLET]: basePerformanceChartDimension,
  [ScreenSize.DESKTOP]: basePerformanceChartDimension,
  [ScreenSize.DESKTOP_HD]: basePerformanceChartDimension,
};

const performanceChartSimplifiedDimension: ChartDimension = {
  width: 0,
  height: 290,
  margin: {
    top: 10,
    right: 0,
    bottom: 30,
    left: 30,
  },
};

const performanceChartSimplifiedDimensionByScreenSize: Record<ScreenSize, ChartDimension> = {
  [ScreenSize.MOBILE]: performanceChartSimplifiedDimension,
  [ScreenSize.TABLET]: performanceChartSimplifiedDimension,
  [ScreenSize.DESKTOP]: performanceChartSimplifiedDimension,
  [ScreenSize.DESKTOP_HD]: performanceChartSimplifiedDimension,
};

export function usePerformanceChartDimension(parentWidth = 0): ChartDimensionWithExtras {
  const screenSize = useScreenSize();

  return React.useMemo(() => {
    const dimensionBaseWidth = performanceChartDimensionByScreenSize[screenSize];
    const dimensionParentWidth = {
      ...dimensionBaseWidth,
      width: parentWidth,
    };

    return getChartDimensionWithExtras(dimensionParentWidth);
  }, [parentWidth, screenSize]);
}

export function usePerformanceSimplifiedChartDimension(parentWidth = 0): ChartDimensionWithExtras {
  const screenSize = useScreenSize();

  return React.useMemo(() => {
    const dimensionBaseWidth = performanceChartSimplifiedDimensionByScreenSize[screenSize];
    const dimensionParentWidth = {
      ...dimensionBaseWidth,
      width: parentWidth,
    };

    return getChartDimensionWithExtras(dimensionParentWidth);
  }, [parentWidth, screenSize]);
}
