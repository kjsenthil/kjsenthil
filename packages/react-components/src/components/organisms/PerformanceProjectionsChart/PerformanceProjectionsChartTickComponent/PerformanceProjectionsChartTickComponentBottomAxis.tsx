import * as React from 'react';
import { TickRendererProps } from '@visx/axis';
import { Text } from '@visx/text';
import { useBreakpoint, useChartStyles } from '../../../../hooks';

export interface PerformanceProjectionsChartTickComponentBottomAxisProps extends TickRendererProps {
  // Note: we have to pass chartStyles as a prop rather than having
  // PerformanceChartTickComponent uses useChartStyles() directly. Otherwise we
  // will run into the mismatched hook call order error when the chart period
  // changes.
  // The cause for this is likely due to how d3 is handling the showing / hiding
  // of chart ticks. Unknown for sure at this point but passing the chartStyles
  // down as a prop works well enough.
  chartStyles: ReturnType<typeof useChartStyles>;

  todayAge: number;

  // Default mode shows both years and ages.
  // Simplified mode shows 'TODAY' when the year is today's year, and only ages
  // otherwise.
  displayMode: 'default' | 'simplified';

  // By providing the final year, we can shift the final tick to the left to
  // avoid margin collision.
  finalYear?: number;
}

export interface GetAgeTextProps {
  tickYear: number | undefined;
  todayAge: number | undefined;

  // If true, will show the age number only rather than "AGE N". This is used in
  // mobile views.
  showAgeNumberOnly?: boolean;
}

/**
 * The formatted value passed to our tick component will be a full year string.
 * This function adds an age text to the formatted value.
 */
export function getAgeText({ todayAge, tickYear, showAgeNumberOnly }: GetAgeTextProps): string {
  const prefix = showAgeNumberOnly ? '' : 'AGE\u00a0';

  if (
    tickYear === undefined ||
    todayAge === undefined ||
    Number.isNaN(tickYear) ||
    Number.isNaN(todayAge)
  ) {
    return `${prefix}UNKNOWN`;
  }

  const todayYear = new Date().getFullYear();

  const age = todayAge + (tickYear - todayYear);

  return `${prefix}${age}`;
}

export default function PerformanceProjectionsChartTickComponentBottomAxis({
  chartStyles,
  todayAge,
  displayMode,
  finalYear,
  x,
  y,
  formattedValue,
  ...tickLabelProps
}: PerformanceProjectionsChartTickComponentBottomAxisProps) {
  const { isMobile } = useBreakpoint();

  // When the tick is for today's year, then we always display "TODAY".
  const todayYear = new Date().getFullYear();
  const tickYear = Number(formattedValue);
  const text =
    tickYear === todayYear
      ? 'TODAY'
      : getAgeText({
          tickYear,
          todayAge,
          showAgeNumberOnly: isMobile,
        });

  // We move the final tick a bit to the left to avoid it being clipped
  const textDxOffset = finalYear && Number(formattedValue) === finalYear ? -40 : undefined;

  return (
    <Text
      x={x}
      dx={textDxOffset}
      y={y}
      {...tickLabelProps}
      fill={chartStyles.TEXT_COLOR.AXES}
      fontSize={chartStyles.TEXT_SIZE.AXES}
      fontFamily={chartStyles.TEXT_FONT.COMMON}
      width={56}
      lineHeight={15}
      style={{ fontWeight: 'bold' }}
    >
      {text}
    </Text>
  );
}
