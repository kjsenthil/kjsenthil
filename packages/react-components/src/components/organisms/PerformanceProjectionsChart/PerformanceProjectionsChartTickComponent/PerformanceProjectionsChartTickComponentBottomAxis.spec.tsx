import * as React from 'react';
import { renderWithTheme } from '@tsw/test-util';
import PerformanceProjectionsChartTickComponentBottomAxis, {
  getAgeText,
  PerformanceProjectionsChartTickComponentBottomAxisProps,
} from './PerformanceProjectionsChartTickComponentBottomAxis';
import { useChartStyles } from '../../../../hooks';

describe('PerformanceProjectionsChartTickComponentBottomAxis', () => {
  describe('PerformanceProjectionsChartTickComponentBottomAxis component', () => {
    const todayYear = new Date().getFullYear();
    const nextYear = todayYear + 1;
    const todayAge = 30;

    // The TickComponent needs to receive chartStyles as a prop
    const ComponentWithChartStyles = (
      props: Pick<
        PerformanceProjectionsChartTickComponentBottomAxisProps,
        'displayMode' | 'formattedValue'
      >
    ) => {
      const chartStyles = useChartStyles();

      return (
        <PerformanceProjectionsChartTickComponentBottomAxis
          chartStyles={chartStyles}
          x={0}
          y={0}
          todayAge={todayAge}
          {...props}
        />
      );
    };

    test("The tick component renders correctly for display mode 'default'", () => {
      expect(
        renderWithTheme(
          <svg>
            <ComponentWithChartStyles displayMode="default" formattedValue={`${todayYear}`} />
            <ComponentWithChartStyles displayMode="default" formattedValue={`${nextYear}`} />
          </svg>
        )
      ).toMatchSnapshot();
    });

    test("The tick component renders correctly for display mode 'simplified'", () => {
      expect(
        renderWithTheme(
          <svg>
            <ComponentWithChartStyles displayMode="simplified" formattedValue={`${todayYear}`} />
            <ComponentWithChartStyles displayMode="simplified" formattedValue={`${nextYear}`} />
          </svg>
        )
      ).toMatchSnapshot();
    });
  });

  describe('getAgeText function', () => {
    const getAgeTextCases: Array<[string | undefined, number, string]> = [
      [`${new Date().getFullYear()}`, 30, 'AGE\u00a030'],
      [`${new Date().getFullYear() + 10}`, 30, 'AGE\u00a040'],
      [undefined, 30, 'AGE\u00a0UNKNOWN'],
    ];

    test.each(getAgeTextCases)(
      "The function to calculate user's age works correctly when formattedValue is %p and today's age is %i",
      (formattedValue, todayAge, expectedAgeText) => {
        const ageText = getAgeText(formattedValue, todayAge);

        expect(ageText).toBe(expectedAgeText);
      }
    );
  });
});
