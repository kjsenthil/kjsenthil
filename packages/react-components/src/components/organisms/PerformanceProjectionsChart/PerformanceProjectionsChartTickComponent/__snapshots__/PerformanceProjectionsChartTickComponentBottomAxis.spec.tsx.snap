// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PerformanceProjectionsChartTickComponentBottomAxis PerformanceProjectionsChartTickComponentBottomAxis component The tick component renders correctly for display mode 'default' 1`] = `
Object {
  "result": Object {
    "asFragment": [Function],
    "baseElement": <body>
      <div>
        <svg>
          <svg
            font-size="12"
            style="overflow: visible;"
            x="0"
            y="0"
          >
            <text
              fill="#aeaabf"
              font-family="Montserrat,Roboto"
              font-size="12"
              style="font-weight: bold;"
              text-anchor="start"
              transform=""
              x="0"
              y="0"
            >
              <tspan
                dy="0"
                x="0"
              >
                TODAY
              </tspan>
            </text>
          </svg>
          <svg
            font-size="12"
            style="overflow: visible;"
            x="0"
            y="0"
          >
            <text
              fill="#aeaabf"
              font-family="Montserrat,Roboto"
              font-size="12"
              style="font-weight: bold;"
              text-anchor="start"
              transform=""
              x="0"
              y="15"
            >
              <tspan
                dy="0"
                x="0"
              >
                AGE 30
              </tspan>
            </text>
          </svg>
          <svg
            font-size="12"
            style="overflow: visible;"
            x="0"
            y="0"
          >
            <text
              fill="#aeaabf"
              font-family="Montserrat,Roboto"
              font-size="12"
              style="font-weight: bold;"
              text-anchor="start"
              transform=""
              x="0"
              y="0"
            >
              <tspan
                dy="0"
                x="0"
              >
                2022
              </tspan>
            </text>
          </svg>
          <svg
            font-size="12"
            style="overflow: visible;"
            x="0"
            y="0"
          >
            <text
              fill="#aeaabf"
              font-family="Montserrat,Roboto"
              font-size="12"
              style="font-weight: bold;"
              text-anchor="start"
              transform=""
              x="0"
              y="15"
            >
              <tspan
                dy="0"
                x="0"
              >
                AGE 31
              </tspan>
            </text>
          </svg>
        </svg>
      </div>
      <svg
        style="width: 0px; height: 0px; position: absolute; top: -100%; left: -100%;"
      >
        <text
          id="__react_svg_text_measurement_id"
          style="font-weight: bold;"
        >
          AGE 31
        </text>
      </svg>
    </body>,
    "container": <div>
      <svg>
        <svg
          font-size="12"
          style="overflow: visible;"
          x="0"
          y="0"
        >
          <text
            fill="#aeaabf"
            font-family="Montserrat,Roboto"
            font-size="12"
            style="font-weight: bold;"
            text-anchor="start"
            transform=""
            x="0"
            y="0"
          >
            <tspan
              dy="0"
              x="0"
            >
              TODAY
            </tspan>
          </text>
        </svg>
        <svg
          font-size="12"
          style="overflow: visible;"
          x="0"
          y="0"
        >
          <text
            fill="#aeaabf"
            font-family="Montserrat,Roboto"
            font-size="12"
            style="font-weight: bold;"
            text-anchor="start"
            transform=""
            x="0"
            y="15"
          >
            <tspan
              dy="0"
              x="0"
            >
              AGE 30
            </tspan>
          </text>
        </svg>
        <svg
          font-size="12"
          style="overflow: visible;"
          x="0"
          y="0"
        >
          <text
            fill="#aeaabf"
            font-family="Montserrat,Roboto"
            font-size="12"
            style="font-weight: bold;"
            text-anchor="start"
            transform=""
            x="0"
            y="0"
          >
            <tspan
              dy="0"
              x="0"
            >
              2022
            </tspan>
          </text>
        </svg>
        <svg
          font-size="12"
          style="overflow: visible;"
          x="0"
          y="0"
        >
          <text
            fill="#aeaabf"
            font-family="Montserrat,Roboto"
            font-size="12"
            style="font-weight: bold;"
            text-anchor="start"
            transform=""
            x="0"
            y="15"
          >
            <tspan
              dy="0"
              x="0"
            >
              AGE 31
            </tspan>
          </text>
        </svg>
      </svg>
    </div>,
    "debug": [Function],
    "findAllByAltText": [Function],
    "findAllByDisplayValue": [Function],
    "findAllByLabelText": [Function],
    "findAllByPlaceholderText": [Function],
    "findAllByRole": [Function],
    "findAllByTestId": [Function],
    "findAllByText": [Function],
    "findAllByTitle": [Function],
    "findByAltText": [Function],
    "findByDisplayValue": [Function],
    "findByLabelText": [Function],
    "findByPlaceholderText": [Function],
    "findByRole": [Function],
    "findByTestId": [Function],
    "findByText": [Function],
    "findByTitle": [Function],
    "getAllByAltText": [Function],
    "getAllByDisplayValue": [Function],
    "getAllByLabelText": [Function],
    "getAllByPlaceholderText": [Function],
    "getAllByRole": [Function],
    "getAllByTestId": [Function],
    "getAllByText": [Function],
    "getAllByTitle": [Function],
    "getByAltText": [Function],
    "getByDisplayValue": [Function],
    "getByLabelText": [Function],
    "getByPlaceholderText": [Function],
    "getByRole": [Function],
    "getByTestId": [Function],
    "getByText": [Function],
    "getByTitle": [Function],
    "queryAllByAltText": [Function],
    "queryAllByDisplayValue": [Function],
    "queryAllByLabelText": [Function],
    "queryAllByPlaceholderText": [Function],
    "queryAllByRole": [Function],
    "queryAllByTestId": [Function],
    "queryAllByText": [Function],
    "queryAllByTitle": [Function],
    "queryByAltText": [Function],
    "queryByDisplayValue": [Function],
    "queryByLabelText": [Function],
    "queryByPlaceholderText": [Function],
    "queryByRole": [Function],
    "queryByTestId": [Function],
    "queryByText": [Function],
    "queryByTitle": [Function],
    "rerender": [Function],
    "unmount": [Function],
  },
}
`;

exports[`PerformanceProjectionsChartTickComponentBottomAxis PerformanceProjectionsChartTickComponentBottomAxis component The tick component renders correctly for display mode 'simplified' 1`] = `
Object {
  "result": Object {
    "asFragment": [Function],
    "baseElement": <body>
      <svg
        style="width: 0px; height: 0px; position: absolute; top: -100%; left: -100%;"
      >
        <text
          id="__react_svg_text_measurement_id"
          style="font-weight: bold;"
        >
          AGE 31
        </text>
      </svg>
      <div>
        <svg>
          <svg
            font-size="12"
            style="overflow: visible;"
            x="0"
            y="0"
          >
            <text
              fill="#aeaabf"
              font-family="Montserrat,Roboto"
              font-size="12"
              style="font-weight: bold;"
              text-anchor="start"
              transform=""
              x="0"
              y="0"
            >
              <tspan
                dy="0"
                x="0"
              >
                TODAY
              </tspan>
            </text>
          </svg>
          <svg
            font-size="12"
            style="overflow: visible;"
            x="0"
            y="0"
          >
            <text
              fill="#aeaabf"
              font-family="Montserrat,Roboto"
              font-size="12"
              style="font-weight: bold;"
              text-anchor="start"
              transform=""
              x="0"
              y="0"
            >
              <tspan
                dy="0"
                x="0"
              >
                AGE 31
              </tspan>
            </text>
          </svg>
        </svg>
      </div>
    </body>,
    "container": <div>
      <svg>
        <svg
          font-size="12"
          style="overflow: visible;"
          x="0"
          y="0"
        >
          <text
            fill="#aeaabf"
            font-family="Montserrat,Roboto"
            font-size="12"
            style="font-weight: bold;"
            text-anchor="start"
            transform=""
            x="0"
            y="0"
          >
            <tspan
              dy="0"
              x="0"
            >
              TODAY
            </tspan>
          </text>
        </svg>
        <svg
          font-size="12"
          style="overflow: visible;"
          x="0"
          y="0"
        >
          <text
            fill="#aeaabf"
            font-family="Montserrat,Roboto"
            font-size="12"
            style="font-weight: bold;"
            text-anchor="start"
            transform=""
            x="0"
            y="0"
          >
            <tspan
              dy="0"
              x="0"
            >
              AGE 31
            </tspan>
          </text>
        </svg>
      </svg>
    </div>,
    "debug": [Function],
    "findAllByAltText": [Function],
    "findAllByDisplayValue": [Function],
    "findAllByLabelText": [Function],
    "findAllByPlaceholderText": [Function],
    "findAllByRole": [Function],
    "findAllByTestId": [Function],
    "findAllByText": [Function],
    "findAllByTitle": [Function],
    "findByAltText": [Function],
    "findByDisplayValue": [Function],
    "findByLabelText": [Function],
    "findByPlaceholderText": [Function],
    "findByRole": [Function],
    "findByTestId": [Function],
    "findByText": [Function],
    "findByTitle": [Function],
    "getAllByAltText": [Function],
    "getAllByDisplayValue": [Function],
    "getAllByLabelText": [Function],
    "getAllByPlaceholderText": [Function],
    "getAllByRole": [Function],
    "getAllByTestId": [Function],
    "getAllByText": [Function],
    "getAllByTitle": [Function],
    "getByAltText": [Function],
    "getByDisplayValue": [Function],
    "getByLabelText": [Function],
    "getByPlaceholderText": [Function],
    "getByRole": [Function],
    "getByTestId": [Function],
    "getByText": [Function],
    "getByTitle": [Function],
    "queryAllByAltText": [Function],
    "queryAllByDisplayValue": [Function],
    "queryAllByLabelText": [Function],
    "queryAllByPlaceholderText": [Function],
    "queryAllByRole": [Function],
    "queryAllByTestId": [Function],
    "queryAllByText": [Function],
    "queryAllByTitle": [Function],
    "queryByAltText": [Function],
    "queryByDisplayValue": [Function],
    "queryByLabelText": [Function],
    "queryByPlaceholderText": [Function],
    "queryByRole": [Function],
    "queryByTestId": [Function],
    "queryByText": [Function],
    "queryByTitle": [Function],
    "rerender": [Function],
    "unmount": [Function],
  },
}
`;
